lib/plug/request_id.ex

```diff
@@ -34,22 +34,29 @@ defmodule Plug.RequestId do
 
           plug Plug.RequestId, http_header: "custom-request-id"
 
+    * `:assign_as` - The name of the key that will be used to store the
+      discovered or generated request id in `conn.private`. If not provided,
+      the request id will not be stored.
+
+          plug Plug.RequestId, assign_as: :plug_request_id
+
   """
 
   require Logger
   alias Plug.Conn
   @behaviour Plug
 
   @impl true
-  def init(opts) do
-    Keyword.get(opts, :http_header, "x-request-id")
-  end
+  def init(opts), do: opts
 
   @impl true
-  def call(conn, req_id_header) do
+  def call(conn, opts) do
+    header = Keyword.get(opts, :http_header, "x-request-id")
```

Comment:
Let's not change this. Instead, change `init/1` to return `{http_header, assign_as}`. Performance is important in a plug that gets called potentially on *every* request, and getting this option every single time is not a good idea. Thanks!

Reply:
Done.